### PR TITLE
CLI: Exit when user does not select a storybook project type

### PR DESCRIPTION
**Pull request by yannbf**
Closes #

## What I did

Currently, when users try to install Storybook in a directory where no project type is detected, we prompt if they would like to select from a list. However, if the user selects "no", the process will continue and will get into a failure:

image

This PR fixes it by exiting the process successfully and providing link to the docs:

image

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/lib/cli/src/initiate.ts [hunk 1]
```diff
@@ -227,9 +227,15 @@ const projectTypeInquirer = async (
...
+
+    if (manualFramework) {
+      return installStorybook(manualFramework, packageManager, options);
+    }
   }
-  return Promise.resolve();
+
+  logger.log();
+  logger.log('For more information about installing Storybook: https://storybook.js.org/docs');
+  process.exit(0);
```
Comment (yannbf) [inline code/lib/cli/src/initiate.ts:238]: @shilman reminder to check this later, thanks!

**Review by valentinpalkovic — APPROVED** [review]

File: code/lib/cli/src/initiate.ts (hunk 1 above)
Comment (shilman) [reply code/lib/cli/src/initiate.ts:239]: I'm ok with this. I think it's not great for our telemetry, but hopefully it's a small corner case.

File: code/lib/cli/src/project_types.ts [hunk 2]
```diff
@@ -63,8 +63,8 @@ export const SUPPORTED_RENDERERS: SupportedRenderers[] = [
 export enum ProjectType {
   UNDETECTED = 'UNDETECTED',
   UNSUPPORTED = 'UNSUPPORTED',
-  REACT_SCRIPTS = 'REACT_SCRIPTS',
   REACT = 'REACT',
+  REACT_SCRIPTS = 'REACT_SCRIPTS',
```
Comment (shilman) [inline code/lib/cli/src/project_types.ts:67]: What's the story here?

**Review by shilman — APPROVED** [review]

File: code/lib/cli/src/project_types.ts (hunk 2 above)
Comment (yannbf) [reply code/lib/cli/src/project_types.ts:67]: We have a big error count regarding "Storybook does not support react-scripts < 5.0.0" and this error can also be thrown when you choose react-scripts from the list of projects to install, but you don't have react-scripts installed.

I suspect what is influencing this count so much is that people just press enter when they get a list of options, and react-scripts is the first in the list. By changing the order, people will get `react` instead, free of the possibility of crashing with the react-scripts issue.

**Pull request merged**